### PR TITLE
feat(buffers): Smear partition flushes across 10 second period

### PR DESCRIPTION
**Pull request by mattrobenolt**
When they all flush at the same time, it causes a slight stampede every
interval. This should smear them out a bit.

### CODE REVIEW

File: src/sentry/buffer/redis.py [hunk 1]
```diff
@@ -144,8 +146,13 @@ def process_pending(self, partition=None):
         if partition is None and self.pending_partitions > 1:
             # If we're using partitions, this one task fans out into
             # N subtasks instead.
+            now = timezone.now()
             for i in range(self.pending_partitions):
-                process_pending.apply_async(kwargs={'partition': i})
+                process_pending.apply_async(
+                    kwargs={'partition': i},
+                    # Smear over 10 seconds
+                    eta=now + timedelta(seconds=(float(i) / (self.pending_partitions - 1)) * 10),
```
Comment (mattrobenolt) [inline src/sentry/buffer/redis.py]: Fun fact, `countdown` requires an integer in seconds, and you can't use fractions of a second. `eta` allows an explicit `datetime` object down to millisecond precision.

**Comment by mattrobenolt** [comment]
I think I'll shelve this for right now until I come up with something better. It's mostly fine for now.

File: src/sentry/buffer/redis.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/buffer/redis.py]: docs aren't super clear about it either

> countdown (float) – Number of seconds into the future that the task should execute. Defaults to immediate execution.
> eta (datetime) – Absolute time and date of when the task should be executed. May not be specified if countdown is also supplied.

**Review by evanpurkhiser — APPROVED** [review]
makes sense to me ⏲

**Comment by BYK** [comment]
Closing due to staleness and reduced importance as we now avoid using buffers as much as we can.

**Pull request closed without merging**